spring-rabbit/src/main/java/org/springframework/amqp/rabbit/core/RabbitTemplate.java

```diff
@@ -1594,6 +1598,34 @@ public void onMessage(Message message) {
 		}
 	}
 
+	private QueueingConsumer createQueueingConsumer(final String queueName, Channel channel) throws Exception {
+		channel.basicQos(1);
+		final CountDownLatch latch = new CountDownLatch(1);
+		QueueingConsumer consumer = new QueueingConsumer(channel) {
+
+			@Override
+			public void handleCancel(String consumerTag) throws IOException {
+				super.handleCancel(consumerTag);
+				latch.countDown();
```

Comment:
We need to force close the channel if the consume doesn't work.

See `((ChannelProxy) channel).getTargetChannel().close();` below.

You are probably right about the cancel.